making-data-wrangle/creating-dat-shinyapp.R

```diff
@@ -0,0 +1,275 @@
+           "https://nccs-data.urban.org/dl.php?f=bmf/2014/bmf.bm1412.csv", 
+           "https://nccs-data.urban.org/dl.php?f=bmf/2013/bmf.bm1312.csv",
+           "https://nccs-data.urban.org/dl.php?f=bmf/2012/bmf.bm1212.csv",
+           "https://nccs-data.urban.org/dl.php?f=bmf/2011/bmf.bm1112.csv", 
+           "https://nccs-data.urban.org/dl.php?f=bmf/2010/bmf.bm1011.csv", 
+           "https://nccs-data.urban.org/dl.php?f=bmf/2009/bmf.bm0910.csv"
+           )
+
+#get all the data, but only keep the data with unique EIN numbers 
+
+#start with the first link 
+bmf.all <- read_csv(links[1])
+bmf.all <- bmf.all[ , c("EIN", "ZIP5", "FIPS")]
+
+#list of total EIN's 
+ein.total <- bmf.all$EIN
+
+#list of rows to keep 
+keep.rows <- ein.total %in% ein.ours
+
+#data set to merge 
+bmf.all <- bmf.all[keep.rows, ]
+
+
+#now do the other links
+counter = 1
+for(i in 2:length(links)){
+  #upload new one 
+  bmf.i <- read_csv(links[i])
+  bmf.i <- bmf.i[ , c("EIN", "ZIP5", "FIPS")]
+  
+  #get all EIN's 
+  ein.all <- bmf.all$EIN
+  
+  #get new EIN's 
+  ein.new <- bmf.i$EIN
+  
+  #keep new ones 
+  keep.rows <- (! (ein.new %in% ein.all)) & (ein.new %in% ein.ours)
+  
+  #add to data frame 
+  if(sum(keep.rows >0 )){  bmf.all <- rbind(bmf.all, bmf.i[keep.rows, ])}
+  
+  counter <- counter +1
+}
+  
+
+#merge bmf.all and dat.sec by EIN and ensure all FIPS codes have 5 digits 
+dat.with.geo <- merge(dat.sec , bmf.all, by = "EIN") %>%
+  #99 of them dont have FIPS #'s 
+  #filter these out for now. ask Jesse how to deal with these later
+  filter(!is.na(FIPS)) %>%
+  #make as character without a decmil 
+  mutate(FIPS = as.character(round(as.numeric(FIPS)))) %>%
+  #get number of digits 
+  mutate(digit =  stringr::str_length(FIPS)) %>%
+  #fix the ones with 4 digits to be 5 digits
+  mutate(FIPS = case_when(digit == 4 ~ paste0("0", FIPS),
+                          digit == 5 ~ FIPS)) 
+
+
+
+#### Match FIPS Codes with RUCA codes ######
+#get RUCA codes from FIPS codes
+dat.ruca.raw <- readxl::read_excel("data-raw/cencus-ruca-2010-revised.xlsx", 
+                               skip = 1)
+
+#format so it can be merged 
+dat.ruca <- dat.ruca.raw %>%
+  #select only the columns we care about
+  select("State-County FIPS Code",  "Primary RUCA Code 2010" )%>%
+  #rename them to something readable 
+  rename( FIPS = "State-County FIPS Code", #state-county code 
+          RUCA = "Primary RUCA Code 2010") %>%
+  #get only the ones in our data set
+  filter((FIPS %in% unique(dat.with.geo$FIPS))) %>%
+  #group by FIPS.SC and take the avearge RUCA number 
+  group_by(FIPS) %>%
+  summarise(RUCA = mean(RUCA), .groups = "keep")
+
+
+#merge 
+dat.merge <- merge(dat.with.geo, dat.ruca, by = "FIPS") %>%
+  #make RUCA into rual, suburban, and metropolitin 
+  mutate(LocationType = case_when(RUCA <= 3 ~ "Urban", 
+                                  RUCA > 3 & RUCA < 8 ~ "Suburban", 
+                                  RUCA >= 8 ~ "Rural")) %>%
```

Comment:
I would put a check/ assert here to make sure the merge was right and no duplicates are present/ rows are deleted while merging